extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java

```diff
@@ -38,23 +38,9 @@ static String escape(Object value) {
         if (Number.class.isAssignableFrom(value.getClass()) || value instanceof Boolean) {
             return value.toString();
         }
-        if (value instanceof Date) {
-            SimpleDateFormat dateFormat = new SimpleDateFormat(ISO_DATE_PATTERN);
-            dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
-            Date dateValue = (Date) value;
-            return "ISODate('" + dateFormat.format(dateValue) + "')";
-        }
-        if (value instanceof LocalDate) {
-            LocalDate dateValue = (LocalDate) value;
-            return "ISODate('" + DateTimeFormatter.ISO_LOCAL_DATE.format(dateValue) + "')";
-        }
-        if (value instanceof LocalDateTime) {
-            LocalDateTime dateValue = (LocalDateTime) value;
-            return "ISODate('" + ISO_DATE_FORMATTER.format(dateValue.atZone(ZoneOffset.UTC)) + "')";
-        }
```

Comment:
I'd rather keep the existing if chain than mutate the parameter to an Instant.
It feel strange to me as this methods accept a lot of different type and you mutate to an instant some of them

Reply:
Yeah, I wasn't so pleased with it either.  But I also was trying to avoid repeating line 43 four times.

Please feel free to modify as you see fit.

Reply:
@jyemin can you update your PR to keep the old if chain so I can merge it ?

Reply:
Done